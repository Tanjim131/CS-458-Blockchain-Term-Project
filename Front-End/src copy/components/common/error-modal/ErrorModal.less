.error-modal {
    top: 0;
    .modal-content-wrapper {
        text-align: center;
        .check-circle-icon {
            font-size: 32px;
            color: #D24B19;
        }
    }
}