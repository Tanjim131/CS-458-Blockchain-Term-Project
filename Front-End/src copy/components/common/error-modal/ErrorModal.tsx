import React, { FC } from "react";
import { Modal, Space } from "antd";
import { CloseCircleOutlined } from "@ant-design/icons";

import "./ErrorModal.less";

/**
 * Interface for Common Error Modal.
 */
interface IErrorModal {
  visible: boolean;
  footer?: any;
  onCancel: () => void;
  children: {};
  className?: string;
}

/**
 * This is used for ErrorModal
 *
 * @usage: <ErrorModal isModalVisible={visible} onCancel={function}/>
 */
const ErrorModal: FC<IErrorModal> = (props: IErrorModal) => {
  const { visible, footer, onCancel, children } = props;

  return (
    <Modal
      visible={visible}
      footer={footer}
      centered={true}
      maskClosable={true}
      onCancel={onCancel}
      className="error-modal"
      destroyOnClose={true}
    >
      <div className="modal-content-wrapper">
        <Space direction={"vertical"} size={"middle"}>
          <CloseCircleOutlined className="check-circle-icon" />
          {children}
        </Space>
      </div>
    </Modal>
  );
};

export default ErrorModal;
