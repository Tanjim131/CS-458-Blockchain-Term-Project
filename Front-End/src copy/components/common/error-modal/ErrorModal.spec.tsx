import React from 'react';
import { shallow } from "enzyme";
import ErrorModal from "./ErrorModal";
import { shallowToJson } from "enzyme-to-json";

describe('ErrorModal Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <ErrorModal
                visible={true}
                footer="footer"
                onCancel={mockFunc}
                children={{}}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })
})