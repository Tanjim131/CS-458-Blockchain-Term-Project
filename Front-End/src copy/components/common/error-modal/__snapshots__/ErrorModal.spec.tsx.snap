// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorModal Component Check if component exists or not and capture snapshot 1`] = `
<Modal
  centered={true}
  className="error-modal"
  confirmLoading={false}
  destroyOnClose={true}
  footer="footer"
  maskClosable={true}
  maskTransitionName="fade"
  okType="primary"
  onCancel={[MockFunction]}
  transitionName="zoom"
  visible={true}
  width={520}
>
  <div
    className="modal-content-wrapper"
  >
    <Space
      direction="vertical"
      size="middle"
    >
      <ForwardRef(CloseCircleOutlined)
        className="check-circle-icon"
      />
      <Component />
    </Space>
  </div>
</Modal>
`;
